### ISSUE TITLE
doc/user: document `options` startup setting

### ISSUE THREAD

**Issue by morsapaes** [MaterializeInc/materialize#18235]
### Documentation request

After #17853, we support parsing `options` in the connection string, like `cluster` or `search_path`. As an example, to connect to Materialize using `quickstart` as the default cluster, you can now use:

`psql "postgres://MZ_USER@MZ_HOST:6875/materialize?sslmode=require&options=--cluster%3Dquickstart`

This should be documented in [`/integrations/sql-clients/`](https://materialize.com/docs/integrations/sql-clients/) ([reference](https://www.postgresql.org/docs/current/app-psql.html)).

**Comment by chaas** [comment]
skunkworks idea: ui form (maybe baked into the docs?) that converts input of option k-v pairs into the to connection string

**Comment by chaas** [comment]
More examples of how docs would help users and the current pain points without docs https://materializeinc.slack.com/archives/C02CB7L4TCG/p1679679931633229

**Comment by chuck-alt-delete** [comment]
I've been playing around and have found this to be more comprehensible:
```bash
# source environment variables
source .env
# Set PG password and options inline with psql command
PGPASSWORD=$MZ_PASSWORD \
PGOPTIONS='--cluster=datacouncil2023 --search_path=shop' \
    psql \
        -U $MZ_USER \
        -h $MZ_HOST \
        -p 6875 \
        -d datacouncil2023
```

**Comment by rjnn** [comment]
A user just bumped into this as they were searching for how to set the cluster upon connection.

**Comment by morsapaes** [comment]
Fixed in #24143.
